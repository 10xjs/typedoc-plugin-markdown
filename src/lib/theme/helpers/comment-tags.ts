import * as Handlebars from 'handlebars';
import { CommentTag } from 'typedoc/dist/lib/models';

export function commentTags(this: CommentTag[]) {
  const md = ['\n'];
  this.forEach(tag => {
    md.push(`**\`${tag.tagName}\`** ${tag.text ? Handlebars.helpers.comment.call(tag.text) : ''}`);
  });
  return md.join('\n\n');
}
