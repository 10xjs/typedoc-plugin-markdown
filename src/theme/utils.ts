import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
