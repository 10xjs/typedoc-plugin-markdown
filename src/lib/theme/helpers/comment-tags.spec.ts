import * as Handlebars from 'handlebars';
import { mocked } from 'ts-jest/utils';

import { Fixture, getFixture, getReflectionByName } from '../../../test/test.utils';
import { commentTags } from './comment-tags';

describe(`commentTags helper`, () => {
  test(`should compile`, () => {
    mocked(Handlebars).helpers.comment = () => {
      return 'Test Comments';
    };
    const data = getReflectionByName(getFixture(Fixture.Comment), 'CommentedClass').comment.tags;
    const result = commentTags.call(data);
    expect(result).toMatchSnapshot();
  });
});
