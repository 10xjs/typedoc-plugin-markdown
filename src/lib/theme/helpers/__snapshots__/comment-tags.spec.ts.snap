// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commentTags helper should compile 1`] = `
"


**\`deprecated\`** 

**\`todo\`** Test Comments"
`;
